clients/src/main/java/org/apache/kafka/common/security/authenticator/LoginManager.java

```diff
@@ -27,40 +28,28 @@
 
 import org.apache.kafka.common.config.SaslConfigs;
 import org.apache.kafka.common.network.LoginType;
-import org.apache.kafka.common.security.JaasUtils;
+import org.apache.kafka.common.security.auth.Login;
+import org.apache.kafka.common.security.kerberos.KerberosLogin;
+import org.apache.kafka.common.utils.Utils;
 
 public class LoginManager {
 
     private static final EnumMap<LoginType, LoginManager> CACHED_INSTANCES = new EnumMap(LoginType.class);
```

Comment:
@ijuma The implementation uses a single login context with multiple login modules to support multiple mechanisms. Since Login is associated with login context rather than login module, one LoginManager per login type is sufficient.